### ISSUE TITLE
Invoke helper?

### ISSUE THREAD

**Issue by henrymazza** [DockYard/ember-composable-helpers#119]
I’ve been using my own version of this and find it very useful. Like:

```
<a {{action (pipe (invoke “save” model) (toggle ‘popUpVisible')}} >Save</a>
```

It should accespt objects or arrays, I still haven’t find any downsides on this approach.

PS: If model gets into an error state the pipe is halted (yey!) and the error can be dealt with on route, controller, or a component that explains those to the user.

**Comment by poteto** [comment]
I quite like this. Would you be able to open a PR for this?

cc @martndemus

**Comment by henrymazza** [comment]
It's a little trickier to generalize than I thought. In my use case `{{invoke 'save' model}}` will call on each object if it happens the model being an Array. But so `{{invoke 'objectAt' 2 model}}` will fail. I could test if function is implemented array level than element level, but it doesn't seem intuitive. Thoughts?

**Comment by poteto** [comment]
What is the current implementation you have? It'll be easier to comment on if we could see it.

**Comment by henrymazza** [comment]
I’ll send the PR later and we can discuss over it, ok? 

-- 
Fabio Mazarotto

On 13 de março de 2016 at 13:45:21, Lauren Tan (notifications@github.com) wrote:

I quite like this. Would you be able to open a PR for this?

cc @martndemus

**Comment by poteto** [comment]
Sure, thanks!

**Comment by henrymazza** [comment]
#121

**Comment by henrymazza** [comment]
Closed by #121
